src/misc/lv_style_gen.h

```diff
@@ -15,229 +15,229 @@ extern "C" {
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_FLEX_TRACK_PLACE;
+extern const lv_style_prop_t _lv_style_const_prop_id_FLEX_TRACK_PLACE;
 void lv_style_set_flex_grow(lv_style_t * style, uint8_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_FLEX_GROW;
+extern const lv_style_prop_t _lv_style_const_prop_id_FLEX_GROW;
 #endif /*LV_USE_FLEX*/
 
 #if LV_USE_GRID
 void lv_style_set_grid_column_dsc_array(lv_style_t * style, const int32_t * value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_COLUMN_DSC_ARRAY;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_COLUMN_DSC_ARRAY;
 void lv_style_set_grid_column_align(lv_style_t * style, lv_grid_align_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_COLUMN_ALIGN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_COLUMN_ALIGN;
 void lv_style_set_grid_row_dsc_array(lv_style_t * style, const int32_t * value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_ROW_DSC_ARRAY;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_ROW_DSC_ARRAY;
 void lv_style_set_grid_row_align(lv_style_t * style, lv_grid_align_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_ROW_ALIGN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_ROW_ALIGN;
 void lv_style_set_grid_cell_column_pos(lv_style_t * style, int32_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_COLUMN_POS;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_COLUMN_POS;
 void lv_style_set_grid_cell_x_align(lv_style_t * style, lv_grid_align_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_X_ALIGN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_X_ALIGN;
 void lv_style_set_grid_cell_column_span(lv_style_t * style, int32_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_COLUMN_SPAN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_COLUMN_SPAN;
 void lv_style_set_grid_cell_row_pos(lv_style_t * style, int32_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_ROW_POS;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_ROW_POS;
 void lv_style_set_grid_cell_y_align(lv_style_t * style, lv_grid_align_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_Y_ALIGN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_Y_ALIGN;
 void lv_style_set_grid_cell_row_span(lv_style_t * style, int32_t value);
-LV_ATTRIBUTE_EXTERN_DATA extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_ROW_SPAN;
+extern const lv_style_prop_t _lv_style_const_prop_id_GRID_CELL_ROW_SPAN;
```

Comment:
Those were changed because of doxygen having issues with the LV_ATTRIBUTE_EXTERN_DATA macro.

As I said in the PR comment, DO NOT MERGE THIS.. I said that because I need the CI build to run to make sure that there are no build errors which is obviously is that was caused by a change I had made. I have to dig through the code in order to figure out the way to pass the doxygen config parameter to process and expand the macros prior to collecting the documentation from the header files.

There is more work that needs to be done to get everything to build 100%, give me another day or two to get the last remaining bits sorted out.

Reply:
as far as the missing docs errors. If you click on actions in GitHub and locate the build for the documentation and click on it. There is going to be an attached artifact. That artifact contains a snapshot of everything that is missing in terms of docstrings. I can set the build to fail for missing documentation but the issue there is so much of LVGL is currently undocumented the only person that would be able to document the missing bits is one of the maintainers of LVGL. It could possibly be that only yourself is going to be able to do it. The build will always fail of that documentation is not finished and instead of lumping that amount of work onto someone like yourself right away I opted to accept the current missing documentations and only have the build fail is something new has been added without a docstring or something has changed and the docstring has not been updated. In order to do that we needed to have a snapshot of the state of the documentation which gets saved as a cache and also set as an artifact for inspection. 


If you as a maintainer make the decision to allow a PR to be merged even tho the doc build is failing because there is a missing docstring you can go into the actions and locate the build and invalidate the cache. The nxt time the documentation gets built the build will pass and the new information on the missing docstrings will then be added to the cached files.

Reply:
Let's separate this into 2 things:
1. There are no `Sphinx` errors: it seems we are pretty close.
2. There are no `Doxygen` error: we are working on it with Vic. (It will take some time for sure)

Reply:
most of the doxygen errors are suppressed and that is what Vic is working on. I have all but a single error/warning sorted out with sphinx. The one error is because there are 2 lvgl.h files and it is pitching a fit about it. I don't know why there are 2 but there is also one that is located in the src directory as well as in the root of the repo.

I managed to get the `LV_ATTRIBUTE_EXTERN_DATA` issue all sorted out so that is now working properly. I did have to update the style gen script to change the heading primer. 

It is building with only that single error about there being more than one lvgl.h file and I have tried every which way I can think of to deal with that and nothing seems to be working. If the file gets deleted the docs will build error and warning free...